crates/turborepo-lib/src/shim.rs

```diff
@@ -46,6 +46,24 @@ fn turbo_version_has_shim(version: &str) -> bool {
     version.major > 1
 }
 
+fn turbo_version_in_range(version: &str, root_path: &Path) -> bool {
+    let version = Version::parse(version).unwrap();
+
+    let version_req_string = match read_turbo_json(root_path) {
+        Ok(turbo_json) => turbo_json.turboVersion,
+
+        // Either not present or it failed to read `turbo.json`.
+        // We'll assume this is *.
+        Err(_) => "*".to_string(),
+    };
+
+    // FIXME: this SemVer matching is "Rust" SemVer matching.
```

Comment:
There are a few comparators that aren't supported but it can be convinced to work.